pkg/server/node.go

```diff
@@ -959,6 +959,7 @@ func (n *Node) setupSpanForIncomingRPC(
 		if br == nil {
 			return
 		}
+		// TODO(irfansharif): How come this span is never Finish()-ed? #58721.
```

Comment:
Because `grpcSpan` comes from `Context`, meaning that whoever put it there is in charge of finishing it.
For newSpan, we return `finishSpan` from this method.